### ISSUE TITLE
spyder-kernels problem with Spyder 5

### ISSUE THREAD

**Issue by Shaimahameed** [spyder-ide/spyder#15091]
on MACOS (macos big sur 11.2.3) previous versions worked smoothly but after updating to spyder 5.0.0  spyder‑kernels not working and cannot be installed using ( conda install spyder‑kernels OR pip install spyder‑kernels )

#ERROR Message :
You have missing dependencies!

Mandatory:
spyder_kernels >=2.0.1,<2.1.0 : 2.0.1 (NOK)

Please install them to avoid this message.

Note: Spyder could work without some of these dependencies, however to have a smooth experience when using Spyder we strongly recommend you to install all the listed missing dependencies.

Failing to install these dependencies might result in bugs. Please be sure that any found bugs are not the direct result of missing dependencies, prior to reporting a new issue.

**Comment by ccordoba12** [comment]
Hey @Shaimahameed, thanks for reporting. Sorry for this problem, it was a mistake on our side and you can safely ignore it.

It'll be fixed in our next version, to be released in a couple of days.

**Comment by Shaimahameed** [comment]
@ccordoba12  thank you for the fast reply i'll be waiting

**Comment by kookma** [comment]
Hi @ccordoba12 
> Hey @Shaimahameed, thanks for reporting. Sorry for this problem, it was a mistake on our side and you can safely ignore that message.
> 
> It'll be fixed in our next version, to be released in a couple of days.

I just want to report I have the same issue on Windows 10 + Python 3.9.3. 
But as you described I ignore the message and Spyder does not complain and I can work! 
If I restart the Spyder I get the error message again!

Thank you for your efforts.

**Comment by zapcity** [comment]
Same issue in Windows 10 + Python 3.8.9. I also have spyder-kernels 2.0.1 installed, all via pip.

**Comment by stonebig** [comment]
in your file "\Lib\site-packages\spyder\dependencies.py", change line 66

from
`SPYDER_KERNELS_REQVER = '>=2.0.1,<2.1.0' `

to
`SPYDER_KERNELS_REQVER = '>=2.0.1;<2.1.0'`

and it will be fine

**Comment by texadactyl** [comment]
For Linux, assuming that you have done a local installation, follow the advice of @stonebig but in this directory:
$HOME/.local/lib/python<VERSION>/site-packages/spyder/

**Comment by real-yfprojects** [comment]
I wonder how that wasn't noticed during testing before the release.

**Comment by stonebig** [comment]
there are integration bugs "with the user" that CI don't/can't catch, it's very frequent.

**Comment by real-yfprojects** [comment]
> there are integration bugs "with the user" that CI don't/can't catch, it's very frequent.

I would definitely be kind of nervous before a _big_ release like 5.0.0 and start up spyder once before releasing. Especially if I merged some bots translation files. Although I can see how the issue happened.

**Comment by texadactyl** [comment]
The army of unpaid volunteers missed something between the 2 packages.  Maybe that file was edited weeks ago.  Distractions like real life happen.  I've commited the same sin in another set of github projects and so has other members of my organisation.

**Comment by Jacques2101** [comment]
> For Linux using pip, assuming that you have done a local installation, follow the advice of @stonebig but in this directory:
> $HOME/.local/lib/python`<VERSION>`/site-packages/spyder/

and on macOS where is it ? 
I installed Spyder 5 on a local basis but can't find this file. I found a lot of dependencies.py files (in users/opt/anaconda3) so I suppose they are from my previous installation with anaconda. 
I also tried to find it in .lib/site-packages but can't find Spyder root after 


Thx for your help.

**Comment by kookma** [comment]
> in your file "\Lib\site-packages\spyder\dependencies.py", change line 66
> 
> from
> `SPYDER_KERNELS_REQVER = '>=2.0.1,<2.1.0' `
> 
> to
> `SPYDER_KERNELS_REQVER = '>=2.0.1;<2.1.0'`
> 
> and it will be fine

@stonebig after applying the above correction I get the new error as below (Windows 10 + Python 3.9.3)

```
\lib\site-packages\setuptools\distutils_patch.py:25: UserWarning: Distutils was imported before Setuptools. This usage is
discouraged and may exhibit undesirable behaviors or errors. Please use Setuptools' objects directly or at least import 
Setuptools first.
 warnings.warn(
```

**Comment by real-yfprojects** [comment]
> @stonebig after applying the above correction I get the new error as below (Windows 10 + Python 3.9.3)
> 
> ```
> \lib\site-packages\setuptools\distutils_patch.py:25: UserWarning: Distutils was imported before Setuptools. 

Do you have the newest version for these packages? That can lead to errors.

**Comment by kookma** [comment]
> Do you have the newest version for these packages? That can lead to errors.

I updated the setuptools after reading your comment! While I have recently installed Python 3.9.3+Spyder 5.0, it seems my setuptools- was 49.2.154.2.0 after updating using `` it is now setuptools-54.2.0

The error is gone now! In summary for my case
1. I applied the @stonebig correction https://github.com/spyder-ide/spyder/issues/15091#issuecomment-812848088
2. I updated my setuptools using `pip install -U setuptools`
3. Spyder 5 now works without any error!

Thank you @real-yfprojects and @stonebig

**Comment by byronformwalt** [comment]
> on MACOS (macos big sur 11.2.3) previous versions worked smoothly but after updating to spyder 5.0.0 spyder‑kernels not working and cannot be installed using ( conda install spyder‑kernels OR pip install spyder‑kernels )
> 
> #ERROR Message :
> You have missing dependencies!
> 
> Mandatory:
> spyder_kernels >=2.0.1,<2.1.0 : 2.0.1 (NOK)
> 
> Please install them to avoid this message.
> 
> Note: Spyder could work without some of these dependencies, however to have a smooth experience when using Spyder we strongly recommend you to install all the listed missing dependencies.
> 
> Failing to install these dependencies might result in bugs. Please be sure that any found bugs are not the direct result of missing dependencies, prior to reporting a new issue.

The same issue occurs for a clean install on macOS 11.2.3 using the DMG installer from http://docs.spyder-ide.org/current/installation.html.

**Comment by gokhalen** [comment]
I have the same issue after installing Spyder 5.0.0 on Windows 10 using Anaconda.

**Comment by Kochise** [comment]
> in your file "\Lib\site-packages\spyder\dependencies.py", change line 66
> 
> from
> `SPYDER_KERNELS_REQVER = '>=2.0.1,<2.1.0' `
> 
> to
> `SPYDER_KERNELS_REQVER = '>=2.0.1;<2.1.0'`
> 
> and it will be fine

Why not just `SPYDER_KERNELS_REQVER = '>=2.0.1'` ?

**Comment by real-yfprojects** [comment]
> Why not just `SPYDER_KERNELS_REQVER = '>=2.0.1'` ?

See PR #15100.

**Comment by byronformwalt** [comment]
Thank you for the tip.  When using the DMG-based installer on macOS, it looks like the path to the affected file is actually as follows:

```/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/dependencies.py```

**Comment by ZedTDean** [comment]
it is version 5.0.2 and the problem still there, and changing the , to ; does not work

**Comment by ccordoba12** [comment]
This problem was fixed a month ago in 5.0.1 but unfortunately the Anaconda team hasn't updated Spyder yet, even after asking them to do it several times.
